src/Raven.Server/Documents/TcpHandlers/SubscriptionConnection.cs

```diff
@@ -476,6 +496,91 @@ private IDisposable RegisterForNotificationOnNewDocuments(SubscriptionCriteria c
+            var startChangeVectorAsDictionary = startChangeVector.ToDictionary(x => x.DbId, x => x.Etag);
+
+            if (startChangeVectorAsDictionary.TryGetValue(dbId, out long tempStartEtag))
+                startEtag = tempStartEtag;
+
+            var isLastChangeVectorEqualsToChangeVectorMatchingDatabasesEtag = true;
+
+            // first, try to get the first document by etag, in case we continued from the point we stopped
+            if (startEtag != 0)
+            {
+                using (dbContext.OpenReadTransaction())
+                {
+                    var doc = TcpConnection.DocumentDatabase.DocumentsStorage.GetDocumentsFrom(dbContext, startEtag, 0, 1).FirstOrDefault();
+
+                    if (doc != null)
+                    {
+                        if (startChangeVector.Length == doc.ChangeVector.Length)
+                        {
+                            foreach (var changeVectorItem in doc.ChangeVector)
+                            {
+                                if (startChangeVectorAsDictionary.TryGetValue(changeVectorItem.DbId, out long curEtag) == false
+                                    || changeVectorItem.Etag != curEtag)
+                                {
+                                    isLastChangeVectorEqualsToChangeVectorMatchingDatabasesEtag = false;
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            if (isLastChangeVectorEqualsToChangeVectorMatchingDatabasesEtag == false)
+            {
+                startEtag = 0;
+                using (dbContext.OpenReadTransaction())
+                {
+                    var highestEtag = TcpConnection.DocumentDatabase.DocumentsStorage.GetLastDocumentEtag(dbContext, criteria.Collection);
+
+                    long low = 0, midpoint = 0;
+                    long high = highestEtag;
+
+                    while (low <= high)
+                    {
+                        midpoint = low + (high - low) / 2;
+
+                        var curDocument = TcpConnection.DocumentDatabase.DocumentsStorage.GetDocumentsFrom(dbContext, midpoint).FirstOrDefault();
+
+
+                        var conflictStatus = ConflictsStorage.GetConflictStatus(
+                            remote: curDocument.ChangeVector,
+                            local: startChangeVector);
+
+                        // meaning startCHangeVector >= curDocument.Changevector
+                        if (conflictStatus == ConflictsStorage.ConflictStatus.AlreadyMerged)
+                        {
+                            // as long as startChangeVector is >= then current document's change vector, we take it
+                            startEtag = curDocument.Etag;
+                            if (curDocument.ChangeVector.EqualTo(startChangeVector))
+                                low = high + 1; // stop here
```

Comment:
Better to break / return here, than try to be smart with the low variable. It'll be clearer.